File: leetcode/0001-two-sum/leetcode-0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> seen;
        
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            int diff = target - nums[i];
            if (seen.count(diff)) {  
                return {seen[diff], i};
            } else {
                seen[nums[i]] = i;
            }
        }

        // No solution, which is not allowed by the problem definition
        return {0};
    }
};

File: leetcode/0001-two-sum/leetcode-0001-two-sum.py
class Solution:

    # Finds the known-to-exist unique, distinct indices <i>, <j> such that:
    # <nums[i] + nums[j] = target>
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        # Dict of found differences between <target> and the elements of <nums>
        seen = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in seen:
                return [i, seen[diff]]
            else:
                seen[nums[i]] = i
